### DISCUSSION TITLE
[Proposal] File scoped Namespace like declaration of Class

### DISCUSSION THREAD

**Discussion by proinfocus** [dotnet/csharplang#6217] [Language Ideas] [answered]
# Class Structure
I propose that C# can easily bring in the feature where the Class declaration can be presented in similar way as the file scoped Namespace is done in the newer versions of the language.

To demonstrate it, I have two parts defined below like how it looks today and how it can be done going forward:
## Today
```
namespace Solution.Project;

public class Person
{

	public string FirstName { get; set; }
	public string LastName { get; set; }
	public int Age { get; set; }
}

internal class PersonExtended
{
	public string FirstName { get; private set; }
	public string LastName { get; private set; }
	public int Age { get; private set; }

	PersonExtended (string firstName, string lastName, int age)
	{
		FirstName = firstName;
		LastName = lastName;
		age = Age;
	}
}

```

## Proposed
```

namespace Solution.Project;

public class Person;

public string FirstName { get; set; }
public string LastName { get; set; }
public int Age { get; set; }


internal class PersonExtended;

public string FirstName { get; private set; }
public string LastName { get; private set; }
public int Age { get; private set; }

PersonExtended (string firstName, string lastName, int age)
{
	FirstName = firstName;
	LastName = lastName;
	age = Age;
}

```

Editor/Compiler can easily infer the braces { and } just starting below the line where `class` keyword is defined and to the end of file, if no other `class` keyword is defined for another set of class.

**Comment by ufcpp** [comment, marked as answer]
https://github.com/dotnet/csharplang/discussions/928 https://github.com/dotnet/csharplang/discussions/5785
